pkg/ebpf/common.go

```diff
@@ -103,7 +103,9 @@ func verifyOSVersion(kernelCode uint32, platform string, exclusionList []string)
 		return true, ""
 	}
 
-	return false, fmt.Sprintf("some required functions are missing: %s", strings.Join(missing, ", "))
+	errMsg := fmt.Sprintf("Kernel:%s Platform:%s ", kernelCodeToString(kernelCode), platform)
```

Comment:
I think we're technically already logging these things here: https://github.com/DataDog/datadog-agent/blob/master/cmd/system-probe/main.go#L127-L134

We might just need to move those lines above https://github.com/DataDog/datadog-agent/blob/master/cmd/system-probe/main.go#L116 so that it actually _always_ gets called

Reply:
Ah - the kernel string is excluded above. You can leave the platform out and keep only the kernel version.

Reply:
Two things about lines  https://github.com/DataDog/datadog-agent/blob/master/cmd/system-probe/main.go#L127-L134:

1. they print platform version but *not* the kernel version
2. the check and the error message kind of gets repeated in https://github.com/DataDog/datadog-agent/blob/master/pkg/ebpf/common.go#L63-L66

To address (1), we can get the kernel version in that code block and print that information along with platform. But then we will have to handle the `nil` case and we will end up repeating lines https://github.com/DataDog/datadog-agent/blob/master/pkg/ebpf/common.go#L58-L61.

The other option is to print the kernel version after  https://github.com/DataDog/datadog-agent/blob/master/pkg/ebpf/common.go#L61. 

If we do that, we might as well as print the platform there after https://github.com/DataDog/datadog-agent/blob/master/pkg/ebpf/common.go#L66. That way, we would be able to get rid of lines https://github.com/DataDog/datadog-agent/blob/master/cmd/system-probe/main.go#L127-L134 completely, resulting in less repeated code.

@sunhay, thoughts?

Reply:
🤦‍♂ I didn't refresh the page before writing my comments.

Reply:
Good point:

- Lets move `log.Infof("running system-probe with version: %s", versionString(", "))` up in `main.go` so it's always called (even on error).
- Lets get rid of the platform log from system-probe's `main.go` and move those messages to `IsTracerSupportedByOS()`.
- For kernel version specifically, lets keep it in the error message described in my comment below.

That should remove most if not all the duplication, right?

Reply:
Right, it should remove all the duplication. Will send out an update soon.